lib/http/endpoint.js

```diff
@@ -223,6 +223,10 @@ const defaultResultWriter = (result, request, response, next) => {
 // error thrown upstream that is of our own internal format, this handler does
 // the necessary work to translate that error into an HTTP error and send it out.
 const defaultErrorWriter = (error, request, response) => {
+  if (error instanceof URIError) {
```

Comment:
I like the idea of returning a proper Problem if a `URIError` is thrown for a badly encoded URL. However, if a `URIError` is thrown for another reason, I wouldn't want to swallow it here. What do you think about confirming here that the URL was badly encoded?

```suggestion
  if (error instanceof URIError && urlDecode(request.url).isEmpty()) {
```

Reply:
I think this is a good idea, but I wish there was a more explicit way to check if the error came from express's internals.

Reply:
Some properties of the error, which we could check:

```js
> e.status === 400
> e.statusCode === 400
> e.message.match(/^Failed to decode param '.*'$/)
```

See: https://github.com/expressjs/express/blob/4.x/lib/router/layer.js#L171-L177

Reply:
I've added checks and comments, which may address the concerns.